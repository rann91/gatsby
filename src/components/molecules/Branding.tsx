/* @jsx jsx */
import { Link } from 'gatsby'
import { Box, Flex, Image, jsx, Text } from 'theme-ui'
import useSiteMetadata from '../../hooks/use-site-metadata'
import logo from '../../images/logo.svg'

const Branding = () => {
  const { title } = useSiteMetadata()

  return (
    <Link
      data-testid="branding"
      sx={{ color: 'inherit', textDecoration: 'none' }}
      to="/">
      <Flex sx={{ alignItems: 'center' }}>
        <Box mr={3} sx={{ flexShrink: 0, lineHeight: 0 }}>
          <Image variant="logo" src={logo} alt="Branding Logo" />
        </Box>
        <Text variant="siteTitle">{title}</Text>
      </Flex>
    </Link>
  )
}

export default Branding
