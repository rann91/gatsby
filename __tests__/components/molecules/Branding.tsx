import { render } from '@testing-library/react'
import { useStaticQuery } from 'gatsby'
import React from 'react'
import Branding from '../../../src/components/molecules/Branding'
import siteMetadata from '../../fixtures/site-metadata.json'

beforeEach(() => {
  ;(useStaticQuery as jest.Mock).mockImplementationOnce(() => siteMetadata)
})

describe('Branding', () => {
  it('renders correctly', () => {
    const { container } = render(<Branding />)

    expect(container).toMatchSnapshot()
  })
})
