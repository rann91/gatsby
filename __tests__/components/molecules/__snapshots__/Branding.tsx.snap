// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branding renders correctly 1`] = `
<div>
  <a
    class="css-13dgcig-Branding"
    data-testid="branding"
    href="/"
  >
    <div
      class="css-18ov0hz-Branding"
    >
      <div
        class="css-fhum7v-Branding"
      >
        <img
          alt="Branding Logo"
          class="css-9taffg"
          src="test-file-stub"
        />
      </div>
      <div
        class="css-vurnku"
      >
        Gatsby
      </div>
    </div>
  </a>
</div>
`;
